src/careamics/careamist.py

```diff
@@ -643,16 +634,41 @@ def predict(
                 # record datamodule
                 self.pred_datamodule = datamodule
 
-                return self.trainer.predict(
-                    model=self.model, datamodule=datamodule, ckpt_path=checkpoint
-                )
-
             else:
                 raise ValueError(
                     f"Invalid input. Expected a CAREamicsWood instance, paths or "
                     f"np.ndarray (got {type(source)})."
                 )
 
+        # TODO: tidy up: move to a seperate function
+        predictions = self.trainer.predict(
+            model=self.model, datamodule=self.pred_datamodule, ckpt_path=checkpoint
+        )
+        # not sure if necessary
+        if len(predictions) == 0:
+            return predictions
+        is_tiled = len(predictions[0]) == 2
+        if is_tiled:
+            # "de-collate"
+            tile_infos = [
+                tile_info
+                for _, tile_info_list in predictions
+                for tile_info in tile_info_list
+            ]
+            predictions = torch.concatenate([preds for preds, _ in predictions])
+            # convert to numpy
+            predictions = predictions.cpu().numpy()
+            predictions = stitch_prediction(predictions, tile_infos)
+        else:
+            # convert to numpy
+            predictions = [pred.cpu().numpy() for pred in predictions]
+
+        # don't return list if single image
+        if len(predictions) == 1:
```

Comment:
Ignoring the previous comment: would it not make sense for the API to be clearer and always return a list of arrays?

Currently (and even more with the previous comment), users will have to check whether they received a list or a single numpy array. I would find this annoying.

I don't have a super strong opinion here.

Reply:
I added this because it was the previous behaviour of the CAREamics prediction loop, but I am happy to remove! or think about input -> output shapes and types a bit harder

Reply:
Yes I figured!